pxd.c

```diff
@@ -1323,15 +1329,17 @@ static void pxd_free_disk(struct pxd_device *pxd_dev)
 	}
 }
 
-struct pxd_device* find_pxd_device(struct pxd_context *ctx, uint64_t dev_id)
+struct pxd_device* find_pxd_device(struct pxd_context *ctx, uint64_t dev_id, bool *stale)
 {
 	struct pxd_device *pxd_dev_itr, *pxd_dev;
 
 	pxd_dev = NULL;
+	*stale = false;
 	spin_lock(&ctx->lock);
 	list_for_each_entry(pxd_dev_itr, &ctx->list, node) {
 		if (pxd_dev_itr->dev_id == dev_id) {
 			pxd_dev = pxd_dev_itr;
+			*stale = pxd_dev->removing;
```

Comment:
The ctx->lock protects the list, do we need to the hold the lock when removing is set or checked? 
It looks like we grab the pxd_dev->lock while setting this instead. Although its grabbed with the ctx->lock held too.
I dont think the code change is wrong but is confusing as there seems to be implicit assumption that when ctx->lock is help the pxd_dev->lock is also held when updating removing. 
If possible can you add some comments to the code where these both locks are defined so that we know what these locks are supposed to protect?

Reply:
can we fix this implicit assumption of the lock and grab the pxd_dev->lock when we look at removing field?

Reply:
added lock for checking `removing`